src/kixi/hecuba/minorca.clj

```diff
@@ -95,30 +126,54 @@
 (defn prepare-measurements-for-upload
   "Using the mapping file to add the measurements to the
   right embed properties."
-  [input-file mapping-file]
-  (let [input-data (open-input-file input-file)
-        formatted-input (format-input-data input-data)
+  [input-data mapping-file]
+  (let [formatted-input (format-input-data input-data)
         mapping-data (open-input-file mapping-file)
         mapping-ids (format-mapping-data mapping-data)]
     (merge-data-ids formatted-input mapping-ids)))
 
+
+;; Second step: processing and uploading the data
 (defn upload-measurement-data
   "Use the previous function to pre-format the data
   before using helper functions to format more and
   send the POST request."
-  [input-file mapping-file base-url username password]
+  [input-data mapping-file base-url username password]
   (->> (prepare-measurements-for-upload
-        input-file mapping-file)
+        input-data mapping-file)
        (map (fn [[map-ids vec-measure]]
               (api/decide-upload (:entity_id map-ids) (:device_id map-ids)
                                  vec-measure
                                  base-url username password)))))
 
 
+
+;; Bring all the steps together:
+;; Step 0: get the data from the S3 bucket
+;; Step 1: pre-process the data (+ update houses mapping file)
+;; Step 2: process + upload the data
+;; Note: create a file to store the data files processed
 (defn main
   "To do all the things."
-  [input-file mapping-file project_id base-url username password]
-  (pre-processing input-file mapping-file project_id
-                  base-url username password)
-  (upload-measurement-data input-file mapping-file
-                           base-url username password))
+  [project_id base-url username password]
+  (let [{:keys [bucket-name bucket-cred mapping-file processed-file]} config-info
+        s3-files (list-files-in-bucket bucket-name bucket-cred)]
+    (map (fn [f]
+           (let [file-info (s3/get-object bucket-cred bucket-name f)
```

Comment:
This leaks file sockets. `get-object` needs to be called in a way that closes the input-stream (see `with-open`) Same problem with amazonica.